jax/_src/scipy/linalg.py

```diff
@@ -434,3 +435,141 @@ def block_diag(*arrs):
     acc = lax.pad(acc, dtype.type(0), ((0, 0, 0), (0, c, 0)))
     acc = lax.concatenate([acc, a], dimension=0)
   return acc
+
+
+# TODO(phawkins): use static_argnames when jaxlib 0.1.66 is the minimum and
+# remove this wrapper.
+@_wraps(scipy.linalg.eigh_tridiagonal)
+def eigh_tridiagonal(d, e, tol=None, eigvals_only=False):
+  return _eigh_tridiagonal(d, e, tol, eigvals_only)
+
+@partial(jit, static_argnums=(3,))
+def _eigh_tridiagonal(d, e, tol, eigvals_only):
+  if not eigvals_only:
+    raise NotImplementedError("Calculation of eigenvectors is not implemented")
+
+  alpha = d
+  beta = e
+
+  def _sturm(alpha, beta_sq, pivmin, alpha0_perturbation, x):
+    """Implements the Sturm sequence recurrence."""
+    n = alpha.shape[0]
+    zeros = jnp.zeros(x.shape, dtype=jnp.int32)
+    ones = jnp.ones(x.shape, dtype=jnp.int32)
+
+    # The first step in the Sturm sequence recurrence
+    # requires special care if x is equal to alpha[0].
+    def sturm_step0():
+      q = alpha[0] - x
+      count = jnp.where(q < 0, ones, zeros)
+      q = jnp.where(alpha[0] == x, alpha0_perturbation, q)
+      return q, count
+
+    # Subsequent steps all take this form:
+    def sturm_step(i, q, count):
+      q = alpha[i] - beta_sq[i - 1] / q - x
+      count = jnp.where(q <= pivmin, count + 1, count)
+      q = jnp.where(q <= pivmin, jnp.minimum(q, -pivmin), q)
+      return q, count
+
+    # The first step initializes q and count.
+    q, count = sturm_step0()
+
+    # Peel off ((n-1) % blocksize) steps from the main loop, so we can run
+    # the bulk of the iterations unrolled by a factor of blocksize.
+    blocksize = 16
+    i = 1
+    peel = (n - 1) % blocksize
+    unroll_cnt = peel
+
+    def unrolled_steps(args):
+      start, q, count = args
+      for j in range(unroll_cnt):
+        q, count = sturm_step(start + j, q, count)
+      return start + unroll_cnt, q, count
+
+    i, q, count = unrolled_steps((i, q, count))
+
+    # Run the remaining steps of the Sturm sequence using an partially
+    # unrolled while loop.
+    unroll_cnt = blocksize
+    def cond(iqc):
+      i, q, count = iqc
+      return jnp.less(i, n)
+    _, _, count = lax.while_loop(cond, unrolled_steps, (i, q, count))
+    return count
+
+  alpha = jnp.asarray(alpha)
+  beta = jnp.asarray(beta)
+  supported_dtypes = (jnp.float32, jnp.float64)
+  if alpha.dtype not in supported_dtypes or beta.dtype not in supported_dtypes:
+    raise TypeError("Only float32 and float64 inputs are supported as inputs "
+                    "to jax.scipy.linalg.eigh_tridiagonal, got "
+                    f"{alpha.dtype} and {beta.dtype}")
+  n = alpha.shape[0]
+  if n == 1:
+    return alpha
+  beta_abs = jnp.abs(beta)
+  beta_sq = beta * beta
+
+  # Compute an interval containing the eigenvalue of T using the Gerschgorin
```

Comment:
```suggestion
  # Compute an interval containing the eigenvalue of T using the Gershgorin
```

Reply:
https://en.wikipedia.org/wiki/Semyon_Aranovich_Gershgorin says this is an accepted transliteration!

Reply:
Okay you win, though the spelling is inconsistent with a comment a few lines down. (Maybe that's good, as all possible transliterations are represented.)